ui/components/component-library/button-link/button-link.tsx

```diff
@@ -41,10 +41,18 @@ export const ButtonLink: ButtonLinkComponent = React.forwardRef(
         startIconProps={{
           size:
             size === ButtonLinkSize.Inherit ? IconSize.Inherit : IconSize.Sm,
+          className:
+            size === ButtonLinkSize.Inherit
+              ? 'mm-button-link--size-inherit__icon'
+              : '',
```

Comment:
Noticed you cannot pass `startIconProps` to `ButtonLink` without overriding this logic so 
```
<ButtonLink
  size={ButtonLinkSize.Inherit}
  endIconProps={{ 'data-testid': 'end-icon' }}
>
  Learn more
</ButtonLink>
``` 
would break this logic.